src/examples/turtlebot4/README.md

```diff
@@ -0,0 +1,138 @@
+      - Ubuntu 24.04 & ros2 jazzy: [link](https://robotecai-my.sharepoint.com/:u:/g/personal/bartlomiej_boczek_robotec_ai/Ebi0O3eWAtRAl3GL8LuEBqIB0-5fnHDCm7yOtcTucpNfzA?e=G6eJKP)
+
+      > **NOTE** If you would like to make changes to the simulation and create your
+      > own binary please follow this [README.md](https://github.com/RobotecAI/ROSCon2024Tutorial/README.md)
+
+   2. Unzip it:
+
+      ```bash
+      cd rai
+      # for Ubuntu 22.04 jammy and ros2 humlbe
+      unzip -d src/examples/turtlebot4/simulation Turtlebot4_jammyhumble_0.0.1.zip
+
+      # for Ubuntu 24.04 noble and ros2 jazzy
+      unzip -d src/examples/turtlebot4/simulation Turtlebot4_noblejazzy_0.0.1.zip
+      ```
+
+2. Clone and install [rai](https://github.com/RobotecAI/rai)
+
+   1. Follow setps from [1. Setting up the workspace](https://github.com/RobotecAI/rai?tab=readme-ov-file#1-setting-up-the-workspace)
+   2. Then follow steps from [2. Build the project](https://github.com/RobotecAI/rai?tab=readme-ov-file#2-build-the-project)
+
+3. Setup your [OpenAI API](https://openai.com/index/openai-api/). Make sure you have set `OPENAI_API_KEY` variable.
+
+4. Configure `rai_whoami_node` (based on ["Your robot identity in RAI"](https://github.com/RobotecAI/rai/blob/development/docs/create_robots_whoami.md) tutorial):
+
+   1. Create `whoami` package for turtlebot4 in `src/examples/turtlebot4`
+
+      ```bash
+      . ./setup_shell.sh
+      poetry run create_rai_ws --name turtlebot4 --destination-directory src/examples
+      ```
+
+   2. Download official turtlebot4 [data sheet](https://bit.ly/3KCp3Du) into
+      `src/examples/turtlebot4_whoami/description/documentation`
+   3. Download [image](https://s3.amazonaws.com/assets.clearpathrobotics.com/wp-content/uploads/2022/03/16113604/Turtlebot-4-20220207.44.png) of turtlebot4 into `src/examples/turtlebot4_whoami/description/images`
+   4. Create robot's identity. Run the `parse_whoami_package`. This will process the documentation, building
+      it into a vector database, which is used by RAI agent to reason about its identity.
+
+      > **NOTE**: Vector database is created using the OpenAI API. Parsing bigger documents
+      > might lead to costs. Embedding model can be configured in
+      > [config.toml](https://github.com/RobotecAI/rai/blob/development/config.toml#L13)
+
+      ```bash
+      poetry run parse_whoami_package src/examples/turtlebot4_whoami/description
+      # you will be asked to press `y` to continue
+      ```
+
+   5. Rebuild the workspace
+      ```bash
+      colcon build --symlink-install
+      ```
+
+   Congratulations! Your `rai_whoami_node` is configured. In the follwing steps
+   your RAI agent will have turtlebot4 personality.
+
+5. Run rai agent:
+
+   ```bash
+   ros2 launch ./src/examples/turtlebot4/turtlebot.launch.xml \
+       game_launcher:=./src/examples/turtlebot4/simulation/TurtleBot4DemoGamePackage/TurtleBot4Demo.GameLauncher
```

Comment:
Ubuntu 22 humble fails with:

```
[rai_whoami_node-1] [INFO] [1729006016.012198943] [rai_whoami_node]: Robot constitution loaded from /home/krachwal/Projects/internal/rai/install/turtlebot4_whoami/share/turtlebot4_whoami/description/robot_constitution.txt
[rai_whoami_node-1] Traceback (most recent call last):
[rai_whoami_node-1]   File "/home/krachwal/Projects/internal/rai/install/rai_whoami/lib/rai_whoami/rai_whoami_node", line 33, in <module>
[rai_whoami_node-1]     sys.exit(load_entry_point('rai-whoami', 'console_scripts', 'rai_whoami_node')())
[rai_whoami_node-1]   File "/home/krachwal/Projects/internal/rai/build/rai_whoami/rai_whoami/rai_whoami_node.py", line 191, in main
[rai_whoami_node-1]     who_am_i_node = WhoAmI()
[rai_whoami_node-1]   File "/home/krachwal/Projects/internal/rai/build/rai_whoami/rai_whoami/rai_whoami_node.py", line 81, in __init__
[rai_whoami_node-1]     self.faiss_index = self._load_documentation()
[rai_whoami_node-1]   File "/home/krachwal/Projects/internal/rai/build/rai_whoami/rai_whoami/rai_whoami_node.py", line 84, in _load_documentation
[rai_whoami_node-1]     faiss_index = FAISS.load_local(
[rai_whoami_node-1]   File "/home/krachwal/.cache/pypoetry/virtualenvs/rai-Qp9irBSo-py3.10/lib/python3.10/site-packages/langchain_community/vectorstores/faiss.py", line 1203, in load_local
[rai_whoami_node-1]     index = faiss.read_index(str(path / f"{index_name}.faiss"))
[rai_whoami_node-1]   File "/home/krachwal/.cache/pypoetry/virtualenvs/rai-Qp9irBSo-py3.10/lib/python3.10/site-packages/faiss/swigfaiss_avx2.py", line 10538, in read_index
[rai_whoami_node-1]     return _swigfaiss_avx2.read_index(*args)
[rai_whoami_node-1] RuntimeError: Error in faiss::FileIOReader::FileIOReader(const char*) at /project/faiss/faiss/impl/io.cpp:67: Error: 'f' failed: could not open /home/krachwal/Projects/internal/rai/install/turtlebot4_whoami/share/turtlebot4_whoami/description/index.faiss for reading: No such file or directory
[ERROR] [rai_whoami_node-1]: process has died [pid 58662, exit code 1, cmd '/home/krachwal/Projects/internal/rai/install/rai_whoami/lib/rai_whoami/rai_whoami_node --ros-args -r __node:=rai_whoami_node --params-file /tmp/launch_params_r1_cikmj'].
```

Reply:
I rebased the branch on the newest changes in whoami from development - please try again